core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -298,6 +305,7 @@ public NodeIndicesStats stats(boolean includePrevious, CommonStatsFlags flags) {
     /**
      * Returns <tt>true</tt> if changes (adding / removing) indices, shards and so on are allowed.
      */
+    @Override
     public boolean changesAllowed() {
```

Comment:
we can rather make it private and throw an exception like `ensureStarted()` and call it before we go anywhere that violates the condition but even that is best effort.